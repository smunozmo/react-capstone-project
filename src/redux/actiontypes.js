export const GET_ALL_CHARACTERS = 'GET_ALL_CHARACTERS';
export const GET_BY_CATEGORIES = 'GET_BY_CATEGORIES';
export const GET_ALL_SPECIES = 'GET_ALL_SPECIES';
export const CLEAR_LIST = 'CLEAR_LIST';
