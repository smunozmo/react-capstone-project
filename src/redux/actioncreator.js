import { GET_ALL_CHARACTERS, GET_BY_CATEGORIES, GET_ALL_SPECIES, CLEAR_LIST } from './actiontypes';

export const getCharacters = (payload) => ({
  type: GET_ALL_CHARACTERS,
  payload,
});

export const getByCategories = (payload) => ({
  type: GET_BY_CATEGORIES,
  payload,
});

export const getAllSpecies = (payload) => ({
  type: GET_ALL_SPECIES,
  payload,
});

export const clearList = (payload) => ({
  type: CLEAR_LIST,
  payload,
});
