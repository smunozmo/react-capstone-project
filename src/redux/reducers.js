import { GET_ALL_CHARACTERS, GET_BY_CATEGORIES, GET_ALL_SPECIES } from './actiontypes';

const initialState = [];

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_ALL_CHARACTERS: {
      const newState = Object.keys(action.payload).map((key) => ({
        total_count: action.payload[key].info.count,
        info: action.payload[key].results,
      }));
      // const newState = action.payload;
      // console.log('ns', newState);
      return newState;
    }
    case GET_BY_CATEGORIES: {
      const newState = {
        category_index: action.payload,
      };
      return [...state, newState];
    }
    case GET_ALL_SPECIES: {
      const newState = {
        species_list: action.payload,
      };
      return [...state, newState];
    }
    default:
      return state;
  }
};

export default reducer;
