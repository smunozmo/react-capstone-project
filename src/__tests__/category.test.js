/* eslint-disable no-unused-vars */
import React from 'react';
import { createStore } from 'redux';
import { cleanup, render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import Categories from '../pages/categories';
import apiresponse from '../__mocks__/api';

afterEach(cleanup);

const startingState = apiresponse;

const reducer = (state = startingState, action) => {
  switch (action.type) {
    case 'GET_ALL_CHARACTERS': {
      const newState = Object.keys(action.payload).map((key) => ({
        total_count: action.payload[key].info.count,
        info: action.payload[key].results,
      }));
      newState.shift();
      return newState;
    }
    case 'GET_BY_CATEGORIES': {
      const newState = {
        category_index: action.payload,
      };
      return [...state, newState];
    }
    case 'GET_ALL_SPECIES': {
      const newState = {
        species_list: action.payload,
      };
      return [...state, newState];
    }
    case 'CLEAR_LIST': {
      return [];
    }
    default:
      return state;
  }
};

function renderWithRedux(
  component,
  { initialState, store = createStore(reducer, initialState) } = {},
) {
  return {
    ...render(<Provider store={store}><BrowserRouter>{component}</BrowserRouter></Provider>),
  };
}

it('Renders with Redux', () => {
  const { getByText } = renderWithRedux(<Categories />);
});

it('Renders "species breakdown" text', () => {
  const { getByText } = renderWithRedux(<Categories />);
  expect(screen.getByText('species breakdown')).toBeInTheDocument();
});

it('Renders data from state', () => {
  const { getByText } = renderWithRedux(<Categories />);
  expect(screen.getByText('Morty Smith')).toBeInTheDocument();
});

it('Home snapshot test', () => {
  const { myrender } = renderWithRedux(<Categories />);
  expect(myrender).toMatchSnapshot();
});
