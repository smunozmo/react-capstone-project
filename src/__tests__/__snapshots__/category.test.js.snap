// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot test 1`] = `undefined`;
